geopandas/geoseries.py

```diff
@@ -203,28 +187,25 @@ def __new__(cls, data=None, index=None, crs=None, **kwargs):
                     # pd.Series with empty data gives float64 for older pandas versions
                     s = s.astype(object)
                 else:
-                    warnings.warn(_SERIES_WARNING_MSG, FutureWarning, stacklevel=2)
-                    return s
+                    raise TypeError(
+                        "Non geometry data passed to GeoSeries constructor, "
+                        f"received data of dtype '{s.dtype}'"
+                    )
             # try to convert to GeometryArray, if fails return plain Series
             try:
                 data = from_shapely(s.values, crs)
             except TypeError:
-                warnings.warn(_SERIES_WARNING_MSG, FutureWarning, stacklevel=2)
-                return s
+                raise TypeError(
+                    "Non geometry data passed to GeoSeries constructor, "
+                    f"received data of dtype '{s.dtype}'"
+                )
             index = s.index
             name = s.name
 
-        self = super(GeoSeries, cls).__new__(cls)
-        super(GeoSeries, self).__init__(data, index=index, name=name, **kwargs)
-
+        # TODO what about GeoSeries from GeoSeries?
+        super().__init__(data, index=index, name=name, **kwargs)
```

Comment:
Does that give a problem? (I think the pandas.Series init should handle that OK)

Reply:
No, it was just a thought I had, for the GeoDataFrame constructor there's an early exit and whether this should be consistent. 
https://github.com/geopandas/geopandas/blob/13bec536192a48866334f3b3833e84817cb0fb65/geopandas/geodataframe.py#L149-L155